src/server.c

```diff
@@ -129,23 +130,61 @@ void serverLogRaw(int level, const char *msg) {
     } else {
         int off;
         struct timeval tv;
-        int role_char;
         pid_t pid = getpid();
 
         gettimeofday(&tv,NULL);
         struct tm tm;
         nolocks_localtime(&tm,tv.tv_sec,server.timezone,server.daylight_active);
-        off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
-        snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
-        if (server.sentinel_mode) {
-            role_char = 'X'; /* Sentinel. */
-        } else if (pid != server.pid) {
-            role_char = 'C'; /* RDB / AOF writing child. */
-        } else {
-            role_char = (server.masterhost ? 'S':'M'); /* Slave or Master. */
+        switch(server.log_timestamp_format) {
+            case LOG_TIMESTAMP_DEFAULT:
+                off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
+                snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+                break;
+
+            case LOG_TIMESTAMP_ISO8601:
+                off = strftime(buf,sizeof(buf),"%Y-%m-%dT%H:%M:%S.",&tm);
+                snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+                char tzbuf[6];
+                strftime(tzbuf, sizeof(tzbuf), "%z", &tm);
+                strncat(buf, tzbuf, sizeof(buf) - strlen(buf) - 1);
+                break;
+
+            case LOG_TIMESTAMP_UNIX:
+                snprintf(buf,sizeof(buf),"%ld",tv.tv_sec);
+                break;
+            default:
+                break;
+        }
+
+        switch (server.log_format) {
+            case LOG_FORMAT_LOGFMT: {
+                char *role;
+                if (server.sentinel_mode) {
+                    role = "sentinel";
+                } else if (pid != server.pid) {
+                    role = "RDB/AOF";
+                } else {
+                    role = (server.masterhost ? "secondary" : "master");
```

Comment:
```suggestion
                    role = (server.masterhost ? "replica" : "master");
```
Is the approved PC term for not a master.